shared/src/lib.rs

```diff
@@ -22,10 +22,19 @@ impl AsRef<Path> for TarDownloadPath {
 }
 
 pub fn untar_to_dir(tar_path: &TarDownloadPath, workspace: &Path) -> Result<(), Error> {
-    let file = fs_err::File::open(tar_path.as_ref()).map_err(Error::FsError)?;
-    let tar = GzDecoder::new(file);
-    let mut archive = Archive::new(tar);
-    archive.unpack(workspace).map_err(Error::FsError)?;
+    fs_err::create_dir_all(workspace).map_err(Error::FsError)?;
+
+    // Shelling out due to https://github.com/alexcrichton/tar-rs/issues/369
+    let mut cmd = Command::new("bash");
```

Comment:
You could do this in one command with `tar -xzf {tar_path} -C {workspace}`

```
-C, --directory=DIR
             Change  to  DIR  before performing any operations.  This option is order-sensitive,
             i.e. it affects all options that follow.
(https://manpages.ubuntu.com/manpages/noble/en/man1/tar.1.html)
```

Reply:
Thanks